### ISSUE TITLE
Working examples using fann2

### ISSUE THREAD

**Issue by hoyleb** [FutureLinkCorporation/fann2#14]
Great wrapper to a wicked NN, however I am having heaps of trouble using getting fann2 to behave.

Please could you add some working python examples, in particular using some advanced features such as 

set_train_stop_function
set_callback

For example, if I want to use the function:
train_on_data()  
then I need to ensure that my input and output features
inp, out 
are lists and lists of lists, for example when constructing the training data object:
train_data.set_train_data(inp, out)

e.g. in the below, consider that _x and _y are N-d  and 1-d numpy arrays, then one must convert things to lists, such as:

		inp, out = [], []
		for i in _x:
			inp.append(list(i))
		for i in _y:
			out.append([i])

		train_data = libfann.training_data()
		train_data.set_train_data(inp, out)

Then one can call
ann.train_on_data(train_data, 10 ,1, 0.1 )  

-- and other examples would be must helpful

**Comment by PDS-123** [comment]
@hoyleb : I agree. I've just started working with this wrapper. Do you have any tips where I can start figuring out how to use the functions defined in FANN? 

Also, are all functions in FANN wrapped?

**Comment by hoyleb** [comment]
Hi PDS, 
I still only know about what is/not wrapped by looking at the python source code.
I've written a wrapper to make FaNN behave a bit like a sci-kit learn routine, which I can share, but it's not beautiful code, and I don't have resources for a few weeks to make it so.

**Comment by PDS-123** [comment]
Ah ok! I'm considering creating an executable in C++ to use the neural net and have python execute it... But that would mean that the executable must be maintained. So, I may shift to neurolab. Btw do you have experience working with PyBrain?

**Comment by hoyleb** [comment]
Yes, PyBrain is very nice as a learning/teaching tool, but it is not fast,
so it not useful for real projects.

You can also try SkyNet, which has a python wrapper (but again is not
sk-lern like). There is also a code called PhotoRaptoR, which has a java
GUI. Both of these codes have astrophysics use-cases, but can be applied
for anything.

Is there any particular reason you want to use NNs? If not, check out
AdaBoost and other decision based methods which are orders of magnitude
faster than NNs, and give comparable (often better) results. Use sci-kit
learn for this, it's very fast and all python, and the documentation is a
dream!

Or if you want to peak into deep learning, check out "graphlab", no-learn,
and then hard-core, pylearn2, tourch, lua.

Ben.

**Comment by PDS-123** [comment]
I'm building a surrogate model for a CFD case with the objective to optimise a design. 

Those are really strong recommendations Ben, will go through them and revert back on which I found best suited for my task :)

Pranav

**Comment by FutureLinkOpenSource** [comment]
We aren't using fann actively any more; however, if there's extra time, we'll add some examples to this package. Thanks for all your comments.

**Comment by FutureLinkOpenSource** [comment]
Also, if you want to provide a patch for any missing functionality, we'll integrate it.
